pkg/controller/ambassadorinstallation/scout.go

```diff
@@ -0,0 +1,87 @@
+// From ambassador/internal/pkg/edgectl/scout.go with some modifications & simplifications.
+
+package ambassadorinstallation
+
+import (
+	"context"
+	"github.com/datawire/ambassador-operator/version"
+	"github.com/datawire/ambassador/pkg/metriton"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+// The Scout structure maintains an index, which is the count of calls
+// to Report, incrementing each call.  This provides a sequence of actions
+// to make it easier to search through the reports.
+// The Reporter is simply the Metriton reporting object.
+type Scout struct {
+	index    int
+	Reporter *metriton.Reporter
+}
+
+// Metadata is simply a key and an untyped value, instances passed in as parameters
+// to (s *Scout) Report
+type ScoutMeta struct {
+	Key   string
+	Value interface{}
+}
+
+// Create a new Scout object, with a parameter stating what the Scout instance
+// will be reporting on.  The Ambassador Operator may be installing, updating,
+// or deleting the Ambassador installation.
+// TODO: @Alvaro please check that this is correct -- do we want to use the same
+// TODO: InstallIDFromFilesystem as was used by edgectl install?  Also,
+// TODO: is version.Version the version number we want to report?
+
+func NewScout(mode string) (s *Scout) {
+	return &Scout{
+		Reporter: &metriton.Reporter{
+			Application: "operator",
+			Version:     version.Version,
+			GetInstallID: func(r *metriton.Reporter) (string, error) {
+				id, err := metriton.InstallIDFromFilesystem(r)
```

Comment:
In the context of a short-lived POD, I do not believe we can safely use the `InstallIDFromFilesystem` as it would generate a new install-id every time the pod restarts, thus we would be unable to correlate reconcile runs in a long-lived install.

This is also slightly up to the product owner to decide on the right strategy here, and how important the uniqueness of the install-id is. We could use strategies similar to Ambassador's own ID generator: https://github.com/datawire/ambassador/blob/9d463b8a974b6e1624c481f1d5b9498a386edb61/python/kubewatch.py#L196

Reply:
I wrote a simple GetInstallID function that just looks at the environment variables for now.  We'll need to think of something more appropriate later.

Reply:
now using amb.GetUID() for a unique identifier.